### DISCUSSION TITLE
Can you generate a schema from the Reflection API?

### DISCUSSION THREAD

**Discussion by zaydek** [open-circle/valibot#733] [Q&A] [answered]
I was reading about the reflection API and I was wondering if you serialize the schema as JSON, is it possible to generate a new Valibot schema from the metadata?

```ts
{
  "kind": "schema",
  "type": "object",
  "expects": "Object",
  "async": false,
  "entries": {
    "bar": {
      "kind": "schema",
      "type": "string",
      "expects": "string",
      "async": false
    }
  }
}
```

Would it be possible to dynamically generate the equivalent of this:

```ts
const MySchema1 = v.object({
  bar: v.string(),
});
```

**Comment by fabian-hiller** [comment, marked as answer]
Yes, this technically works for most schemas, methods, and actions, but I do not have time to implement it at the moment. Feel free to start a community project for this use case.

**Reply by zaydek** [reply]
Thanks. So it's note like I can do v.fromShema(<serialized schema>) but your point is it is possible if you write the code to dynamically construct itself using all the normal primitives I take it. Anyway, thanks for the response!

**Reply by fabian-hiller** [reply]
The idea would be to create a recursive function that checks for the `.type` property and rebuilds the schema that way using our own APIs.

**Comment by gadicc** [comment]
Hi all, I vibe-coded something like this over the weekend.  Let's see where it goes :)

https://github.com/gadicc/valibot-serialize

**Reply by fabian-hiller** [reply]
Thank you so much! This looks amazing! Feel free to add it to our [ecosystem page](https://valibot.dev/guides/ecosystem/) (there is an edit button at the end of the page). I am also interested in featuring your library on social media and Discord! Are you cool with it?

**Reply by gadicc** [reply]
Hey, thanks so much for the kind words and offer 🙏

Let me get back to you about this at the end of the week.  Im using this in my own project successfully, but might want to port some of my project-specific code to the library to make common tasks easier for everyone, and would be nice to make any official announcements after including that too.   And cleaning up the README and docs a little more.

So, thanks for the support and encouragement and I'll be in touch again soon.

**Reply by gadicc** [reply]
Ok there's a nice over-engineered CLI tool now too :)

For ecosystems (thanks), I think probably "Valibot to X", right?  Or "Utilities" ?

Thanks also for offer to feature.  Depending on your available time, could we stagger this?  First mention on Discord, and I'll do my best to be available there for a few days.  And after things stabilize from any feedback, do the social media post?

Thanks again for the support!

**Reply by fabian-hiller** [reply]
That sounds good to me. You library probably fits into "Valibot to X" and "X to Valibot". Feel free to add it to both.
